prov/usnic/src/usdf_mem.c

```diff
@@ -80,13 +80,14 @@ static struct fi_ops usdf_mr_ops = {
 };
 
 int
-usdf_reg_mr(struct fid_domain *domain, const void *buf, size_t len,
+usdf_reg_mr(struct fid *fid, const void *buf, size_t len,
 	   uint64_t access, uint64_t offset, uint64_t requested_key,
 	   uint64_t flags, struct fid_mr **mr_o, void *context)
 {
 	struct usdf_mr *mr;
 	struct usdf_domain *udp;
 	int ret;
+	struct fid_domain *domain = container_of(fid, struct fid_domain, fid);
```

Comment:
Generally agreed.  I only typed the first one off the top of my head, partly under the assumption that Sayantan had another patch coming in the near future.  I probably only should have proposed the second version.